subsys/net/lib/nrf_cloud/coap/src/nrf_cloud_coap.c

```diff
@@ -0,0 +1,489 @@
+			 bool last_block, void *user)
+{
+	LOG_DBG("result_code: %d.%02d, offset:0x%X, len:0x%X, last_block:%d",
+		result_code / 32u, result_code & 0x1f, offset, len, last_block);
+	if (result_code != COAP_RESPONSE_CODE_CONTENT) {
+		loc_err = result_code;
+	} else {
+		loc_err = coap_codec_ground_fix_resp_decode(user, payload, len,
+							    COAP_CONTENT_FORMAT_APP_CBOR);
+	}
+}
+
+int nrf_cloud_coap_location_get(struct nrf_cloud_rest_location_request const *const request,
+				struct nrf_cloud_location_result *const result)
+{
+	__ASSERT_NO_MSG(request != NULL);
+	__ASSERT_NO_MSG((request->cell_info != NULL) || (request->wifi_info != NULL));
+	__ASSERT_NO_MSG(result != NULL);
+	CHECK_AUTH();
+	size_t len = sizeof(buffer);
+	int err;
+
+	err = coap_codec_ground_fix_req_encode(request->cell_info, request->wifi_info, buffer, &len,
+					       COAP_CONTENT_FORMAT_APP_CBOR);
+	if (err) {
+		LOG_ERR("Unable to encode cell pos data: %d", err);
+		return err;
+	}
+	err = nrf_cloud_coap_fetch("loc/ground-fix", NULL, buffer, len,
+				   COAP_CONTENT_FORMAT_APP_CBOR,
+				   COAP_CONTENT_FORMAT_APP_CBOR, true,
+				   get_location_callback, result);
+
+	if (!err && !loc_err) {
+		LOG_DBG("Location: %d, %.12g, %.12g, %d", result->type,
+			result->lat, result->lon, result->unc);
+	} else if (err == -EAGAIN) {
+		LOG_ERR("Timeout waiting for location");
+	} else {
+		LOG_ERR("Error getting location; loc_err:%d, err:%d", loc_err, err);
+		err = loc_err;
+	}
+
+	return err;
+}
+
+static int fota_err;
+
+static void get_fota_callback(int16_t result_code,
+		     size_t offset, const uint8_t *payload, size_t len,
+		     bool last_block, void *user)
+{
+	LOG_DBG("result_code: %d.%02d, offset:0x%X, len:0x%X, last_block:%d",
+		result_code / 32u, result_code & 0x1f, offset, len, last_block);
+	if (result_code != COAP_RESPONSE_CODE_CONTENT) {
+		fota_err = result_code;
+	} else if (payload && len) {
+		LOG_INF("Got FOTA response: %.*s", len, (const char *)payload);
+		fota_err = coap_codec_fota_resp_decode(user, payload, len,
+						       COAP_CONTENT_FORMAT_APP_JSON);
+	} else {
+		fota_err = 0;
+	}
+}
+
+int nrf_cloud_coap_fota_job_get(struct nrf_cloud_fota_job_info *const job)
+{
+	__ASSERT_NO_MSG(job != NULL);
+	CHECK_AUTH();
+	int err;
+
+	job->type = NRF_CLOUD_FOTA_TYPE__INVALID;
+
+	err = nrf_cloud_coap_get("fota/exec/current", NULL, NULL, 0,
+				 COAP_CONTENT_FORMAT_APP_CBOR,
+				 COAP_CONTENT_FORMAT_APP_JSON, true, get_fota_callback, job);
+
+	if (!err && !fota_err) {
+		LOG_INF("FOTA job received; type:%d, id:%s, host:%s, path:%s, size:%d",
+			job->type, job->id, job->host, job->path, job->file_size);
+	} else if (!err && (fota_err == COAP_RESPONSE_CODE_NOT_FOUND)) {
+		LOG_INF("No pending FOTA job");
+	} else if (err == -EAGAIN) {
+		LOG_ERR("Timeout waiting for FOTA job");
+	} else {
+		LOG_ERR("Error getting current FOTA job; FOTA rc:%d.%02d, err:%d",
+			fota_err / 32u, fota_err & 0x1f, err);
```

Comment:
I'm not sure I understand your question.

In the `nrf_cloud_coap.h` header I mention this:
```
 * @return 0 if the request succeeded, a positive value indicating a CoAP result code,
 * or a negative error number.

Reply:
`or a negative error number.`

so is there a case where you will print a negative error number here instead of a coap result code?
`fota_err / 32u, fota_err & 0x1f,`

Reply:
I see what you mean now.